components/MechPage/FeeMetrics.jsx

```diff
@@ -0,0 +1,216 @@
+    return null;
+  }
+};
+
+const usePersistentSWR = (key, fetcher) =>
+  useSWR(key, fetcher, {
+    revalidateIfStale: false,
+    revalidateOnFocus: false,
+  });
+
+export const FeeMetrics = () => {
+  const { data: metrics, error } = usePersistentSWR(
+    'FeeFlowMetrics',
+    fetchMetrics,
+  );
+
+  const formerData = useMemo(
+    () => [
+      {
+        id: 'total-fees',
+        label: 'Total Agent Fees Collected',
+        value: metrics?.totalFees || 0,
+        color: '#7a9cf7',
+      },
+      {
+        id: 'unclaimed',
+        label: 'Unclaimed Fees',
+        value: metrics?.unclaimedFees || 0,
+        color: '#90a1b9',
+      },
+      {
+        id: 'claimed',
+        label: 'Claimed Fees',
+        value: metrics?.collectedFees || 0,
+        color: '#5fb178',
+      },
+      {
+        id: 'received',
+        label: 'Fees Received',
+        value: metrics?.recievedFees || 0,
+        color: '#68bcce',
+      },
+      {
+        id: 'olas-burned',
+        label: 'OLAS Burned',
+        value: metrics?.olasBurned || 0,
+        color: '#dab2e4',
+      },
+    ],
+    [metrics],
+  );
+
+  const data = [
+    ['From', 'To', '', { role: 'tooltip', type: 'string', p: { html: true } }],
+    [
+      'Total Agent Fees Collected',
+      'Unclaimed Fees',
+      formerData[1].value,
+      `Total Agent Fees Collected → Unclaimed Fees | $${formerData[1].value} (${Number((formerData[1].value / formerData[0].value) * 100).toFixed(2)}%)`,
+    ],
+    [
+      'Unclaimed Fees',
+      '',
+      formerData[1].value,
+      `Total Agent Fees Collected → Unclaimed Fees | $${formerData[1].value} (${Number((formerData[1].value / formerData[0].value) * 100).toFixed(2)}%)`,
+    ],
+    [
+      'Total Agent Fees Collected',
+      'Claimed Fees',
+      formerData[2].value,
+      `Total Agent Fees Collected → Claimed Fees | $${formerData[2].value} (${Number((formerData[2].value / formerData[0].value) * 100).toFixed(2)}%)`,
+    ],
+    // Olas burned branch is represented as 10% for better visual presentation as 1% appears too small on screens
+    [
+      'Claimed Fees',
+      'OLAS Burned',
+      formerData[2].value * 0.1,
+      `Claimed Fees → OLAS Burned | $${formerData[2].value * 0.1} (1%)`,
+    ],
+    [
+      'Claimed Fees',
+      'Fees Received',
+      formerData[2].value * 0.9,
+      `Claimed Fees → Fees Received | $${formerData[2].value * 0.9} (99%)`,
+    ],
+  ];
+
+  const [windowWidth, setWindowWidth] = useState(1280);
+
+  useEffect(() => {
+    const handleResize = () => setWindowWidth(window.innerWidth);
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  const chartSizes = () => {
+    if (windowWidth < 640) return { fontSize: 12, chartHeight: 200 };
+    if (windowWidth < 1024) return { fontSize: 16, chartHeight: 300 };
+    return { fontSize: 20, chartHeight: 400 };
+  };
```

Comment:
would be nice to move it to a custom hook and call it similar to what we have in mono repo, e.g. useScreen
so that it's easier to write code in different apps, if the same functions are used